.NET Aspire Preview 6 (#682)
dotnet/docs-aspire: docs/fundamentals/dashboard/explore.md

---
title: Explore .NET Aspire dashboard
description: Explore the .NET Aspire dashboard features through the .NET Aspire Starter app.
ms.date: 04/23/2024
ms.topic: reference
---

# Explore the .NET Aspire dashboard

In the upcoming sections, you'll discover how to create a .NET Aspire app and embark on the following tasks:

> [!div class="checklist"]
>
> - Investigate the dashboard's capabilities by using the app generated from the project template as explained in the [Quickstart: Build your first .NET Aspire app.](../../get-started/build-your-first-aspire-app.md)
>
> - Delve into the features of the .NET Aspire dashboard app.

The screenshots featured in this article showcase the dark theme. For more details on theme selection, refer to [Theme selection](#theme-selection).

## Dashboard authentication

> When you run a .NET Aspire app host, the orchestrator starts up all the apps dependent resources and then opens a browser window to the dashboard. The .NET Aspire dashboard requires token-based authentication for its users because it displays environment variables and other sensitive information.

Reviewer: ```suggestion
When you run a .NET Aspire app host, the orchestrator starts up all the app's dependent resources and then opens a browser window to the dashboard. The .NET Aspire dashboard requires token-based authentication for its users because it displays environment variables and other sensitive information.
```

When the dashboard is launched from Visual Studio or Visual Studio Code (with the [C# Dev Kit extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit)), the login page is bypassed, and the dashboard opens directly. This is the typical developer <kbd>F5</kbd> experience, and the authentication login flow is automated by the .NET Aspire tooling.

> However, if you start the app host from the command line you'll be presented with the login page. The console window displays a URL that you can click on to open the dashboard in your browser.

Reviewer: ```suggestion
However, if you start the app host from the command line, you're presented with the login page. The console window displays a URL that you can click on to open the dashboard in your browser.
```

:::image type="content" source="media/explore/dotnet-run-login-url.png" lightbox="media/explore/dotnet-run-login-url.png" alt-text=".NET CLI run command output, showing the login URL with token query string.":::

> The URL contains a token query string (with the token value mapped to the `t` name part) that's used to _log in_ to the dashboard. If you're console supports it, you can hold the <kbd>Ctrl</kbd> key and then click the link to open the dashboard in your browser. This method is easier than copying the token from the console and pasting it into the login page. If you end up on the dashboard login page without either of the previously described methods, you can always return to the console to copy the token.

Reviewer: ```suggestion
The URL contains a token query string (with the token value mapped to the `t` name part) that's used to _log in_ to the dashboard. If your console supports it, you can hold the <kbd>Ctrl</kbd> key and then click the link to open the dashboard in your browser. This method is easier than copying the token from the console and pasting it into the login page. If you end up on the dashboard login page without either of the previously described methods, you can always return to the console to copy the token.
```

:::image type="content" source="media/explore/aspire-login.png" lightbox="media/explore/aspire-login.png" alt-text=".NET Aspire dashboard login page.":::

> The login page accepts a token and provides helpful instructions on how to obtain the token as shown in the following screenshot:

Reviewer: ```suggestion
The login page accepts a token and provides helpful instructions on how to obtain the token, as shown in the following screenshot:
```

Selecting the error count badge navigates to the [Structured logs](#structured-logs-page) page with a filter applied to show only the logs relevant to the resource:

:::image type="content" source="media/explore/structured-logs-errors.png" lightbox="media/explore/structured-logs-errors.png" alt-text="A screenshot of the .NET Aspire dashboard Structured logs page, showing a filter applied to show only the logs relevant to the resource.":::

To see the log entry in detail for the error, select the **View** button to open a window below the list with the structured log entry details:

:::image type="content" source="media/explore/structured-logs-errors-view.png" lightbox="media/explore/structured-logs-errors-view.png" alt-text="A screenshot of the .NET Aspire dashboard Structured logs page, showing a lower window with the structured log entry details.":::

For more information and examples of Structured logs, see the [Structured logs page](#structured-logs-page) section.

> ![NOTE]

> > The resources page isn't available if the dashboard is started without a configured resource service. Instead the dashboard will start on the [Structured logs page](#structured-logs-page). This is the default experience when the dashboard is run in standalone-mode without additional configuration.

Reviewer: ```suggestion
> The resources page isn't available if the dashboard is started without a configured resource service. The dashboard starts on the [Structured logs page](#structured-logs-page) instead. This is the default experience when the dashboard is run in standalone mode without additional configuration.
```
